src/s1reader/s1_annotation.py

```diff
@@ -0,0 +1,316 @@
+                TYPE=token_safe[2].lower(),
+                POL=self.polarization.lower())
+
+        if self.kind=='calibration':
+            phrase_xml_to_search='calibration-{PLATFORM}-{MODE}{SUBSWATH}-{TYPE}-{POL}'.format(
+                PLATFORM=token_safe[0].lower(),
+                MODE=token_safe[1].lower(),
+                SUBSWATH=self.subswath,
+                TYPE=token_safe[2].lower(),
+                POL=self.polarization.lower())
+        elif self.kind=='noise':
+            phrase_xml_to_search='noise-{PLATFORM}-{MODE}{SUBSWATH}-{TYPE}-{POL}'.format(
+                PLATFORM=token_safe[0].lower(),
+                MODE=token_safe[1].lower(),
+                SUBSWATH=self.subswath,
+                TYPE=token_safe[2].lower(),
+                POL=self.polarization.lower())
+        elif self.kind=='':
+            phrase_xml_to_search=f'/{phrase_base}'
+        else:
+            raise ValueError(f'Cannot recognize the annotation type {self.kind}')
+
+        return phrase_xml_to_search
+
+
+    def _load_xml_as_et(self): #kind: ['', 'calibration', 'noise']
+        if os.path.isfile(self.path_annotation):
+            if self.path_annotation.endswith('.zip'):
+                zip_in=zipfile.ZipFile(self.path_annotation,'r')
+                phrase_xml_in_zip=self._get_search_phrase()
+
+                #search for the .xml file
+                filename_xml=None
+                for fileinfo in zip_in.filelist:
+                    if (phrase_xml_in_zip in fileinfo.filename) and (fileinfo.filename.endswith('.xml')):
+                        filename_xml=fileinfo.filename
+                        break
+                
+                #load the xml if found.
+                if filename_xml is None:
+                    raise ValueError('Cannot find the annotation file.')
+                else:
+                     self.xml_et=ET.fromstring(zip_in.read(filename_xml))
+                
+                #finalize
+                self._xml_loaded=filename_xml
+                zip_in.close()
+
+
+            elif self.path_annotation.endswith('.xml'):
+                #load the xml as file
+                self.xml_et=ET.parse(self.path_annotation)
+                self._xml_loaded=self.xml_et
+
+                
+
+        elif os.path.isdir(self.path_annotation):
+            phrase_xml_in_dir=self._get_search_phrase()
+            if self.kind=='':
+                list_file_xml=glob.glob(f'{self.path_annotation}/{phrase_xml_in_dir}*.xml')
+            else:
+                list_file_xml=glob.glob(f'{self.path_annotation}/annotation/{self.kind}-{phrase_xml_in_dir}*.xml')
+            
+            if len(list_file_xml)==1:
+                self.xml_et=ET.parse(list_file_xml[0])
+                self._xml_loaded=list_file_xml[0]
+                
+            else:
+                raise ValueError(f'Search for the annotation file does not look right. len={len(list_file_xml)}')
```

Comment:
Please be more descriptive. Maybe we can divide it into the case in which `len(list_file_xml)` is 0 or more than 1